handlers/place.py

```diff
@@ -0,0 +1,39 @@
+import pandas as pd
+import numpy as np
+from opencage.geocoder import OpenCageGeocode
+
+def place_cleanup(df):
+    places = df['place']
+    newValues = []
+    types = {}
+    coordinates = {}
+    key = ''  # enter your key; get key: https://opencagedata.com/dashboard
+    geocoder = OpenCageGeocode(key)
+
+    for x, string in places.iteritems():
+        # checkes whether value is nan
+        if (not string) or string != string:
+            newValues.append(np.nan)
+            continue
+
+        place = places[x].replace('.', '').split(', ')
+        newPlace = []
+        for i, p in enumerate(place):
+            query = place[i]
+            if query not in types:
+                results = geocoder.geocode(query, pretty='1', no_annotation='1')
+                if results == []:
+                    types[query] = ''
+                    coordinates[query] = ''
+                else:
+                    types[query] = results[0]['components']['_type']
+                    coordinates[query] = (results[0]['geometry']['lat'], results[0]['geometry']['lng'])
+
+            if (types[query] == '') and (len(place) == 1):
+                newValues.append(np.nan)
```

Comment:
we are adding `NaN` here, but we are also adding another value at line 37, which means it is possible that we add two elements in one iteration, which would break the order